src/planner.c

```diff
@@ -872,15 +872,31 @@ timescaledb_get_relation_info_hook(PlannerInfo *root, Oid relation_objectid, boo
 
 				if (chunk->fd.compressed_chunk_id > 0)
 				{
+					Relation relation = table_open(relation_objectid, NoLock);
```

Comment:
Could you rename this variable to something more informative? What does relation_objectid refer to : is this the chunk's oid or compressed chunk's oid?

Reply:
In this case, it is the oid of the uncompressed chunk.  I've renamed `relation` to `uncompressed_chunk` to be clearer.